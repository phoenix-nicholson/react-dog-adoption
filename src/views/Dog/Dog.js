import React from 'react';
import { useState, useEffect } from 'react';
import { fetchDogsById } from '../../services/dog';
import DogDetails from '../../components/DogDetails-v2';
import { deleteDog } from '../../services/dog';

export default function Dog(props) {
  const [dog, setDog] = useState([]);
  const [loading, setLoading] = useState(true);
  const id = props.match.params.id;

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogsById(id);
      setDog(data);
      setLoading(false);
    };

    fetchData();
  }, [id]);

  setTimeout(() => {
    setLoading(false);
  }, 800);

  if (loading) {
    return <div>Loading...</div>;
  }

  const handleDelete = async (e) => {
    e.preventDefault();
    await deleteDog(dog);
    if (dog !== '') {
      alert("You've successfully deleted a dog");
    } else {
      alert('Request to database has failed, please try again');
    }
  };

  return (
    <div>
      <DogDetails dog={dog} handleDelete={handleDelete} />
    </div>
  );
}
