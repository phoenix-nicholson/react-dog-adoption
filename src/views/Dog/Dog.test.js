import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import Dog from './Dog';

test('renders dogDetails', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/10']}>
      <Route path="/dogs/:id" component={Dog} />
    </MemoryRouter>
  );

  await screen.findByText('Barton');
  expect(container).toMatchSnapshot();
});
