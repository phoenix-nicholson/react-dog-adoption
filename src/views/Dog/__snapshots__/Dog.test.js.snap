// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dogDetails 1`] = `
<div>
  <div>
    <div>
      <img
        src="https://placedog.net/500?id=6"
      />
      <h1>
        Barton
      </h1>
      <p>
        Age: 
        1
      </p>
      <p>
        Breed: 
        Weimaraner
      </p>
      <p>
        Adores every person he meets. Very social and lots of energy
      </p>
    </div>
  </div>
</div>
`;
